File: examples/prometheus-exec/tier-p/transform.mjs
import fs from 'node:fs';
import path from 'node:path';

const inputPath = path.join(process.env.PROMETHEUS_INPUT_DIR, 'records');
const outputPath = path.join(process.env.PROMETHEUS_OUTPUT_DIR, 'summary.json');
const records = JSON.parse(fs.readFileSync(inputPath, 'utf8')).records;
const summary = {
  count: records.length,
  ids: records.map(record => record.id).sort(),
  maxRisk: Math.max(...records.map(record => record.risk)),
  totalRisk: records.reduce((total, record) => total + record.risk, 0),
};
fs.writeFileSync(outputPath, `${JSON.stringify(summary, null, 2)}\n`, 'utf8');
process.stdout.write(`${JSON.stringify(summary)}\n`);

File: examples/prometheus-exec/tier-p/transform.py
import json
import os
from pathlib import Path


input_path = Path(os.environ["PROMETHEUS_INPUT_DIR"], "records")
output_path = Path(os.environ["PROMETHEUS_OUTPUT_DIR"], "summary.json")
records = json.loads(input_path.read_text(encoding="utf-8"))["records"]
summary = {
    "count": len(records),
    "ids": sorted(record["id"] for record in records),
    "maxRisk": max(record["risk"] for record in records),
    "totalRisk": sum(record["risk"] for record in records),
}
encoded = json.dumps(summary, separators=(",", ":"), sort_keys=True)
output_path.write_text(json.dumps(summary, indent=2, sort_keys=True) + "\n", encoding="utf-8")
print(encoded)
